src/apps/Altinn.AccessManagement/src/Altinn.AccessMgmt.Persistence/Services/RoleService.cs

```diff
@@ -100,4 +109,37 @@ public async Task<IEnumerable<RolePackageDto>> GetPackagesForRole(Guid id)
 
         return rolePackages.Select(t => new RolePackageDto(t));
     }
+
+    private async Task GetLegacyRoleCodeAndUrn(QueryResponse<ExtRole> roles)
+    {
+        var roleLookup = await roleLookupRepository.GetExtended();
+
+        foreach (var role in roles)
+        {
+            if (roleLookup.FirstOrDefault(x => x.RoleId == role.Id && x.Key == "LegacyCode") != null)
```

Comment:
if matcher på RoleId mens assignmenten på linje 121 matcher på Role.Code.

gjør da også 2 iterasjoner/FirstOrDefault i stedet for bare en og så nullsjekke før assignment.

Som vel også kan gjøres som en oneliner
`role.LegacyRoleCode = roleLookup.FirstOrDefault(x => x.RoleId == role.Id && x.Key == "LegacyCode")?.Value;`

Reply:
Ryddet opp i denne metoden og brukte kodeeksempelet ditt. 
`LegacyRoleCode.Value` er i caps. Skal jeg gjøre LegacyUrn til lowercase?